macro/src/lib.rs

```diff
@@ -149,6 +158,16 @@ pub fn derive_mmio(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
 
         unsafe impl derive_mmio::_MmioMarker for #wrapper_ident<'_> {}
 
+        /// Send is unsafely implemented because sending a register block pointer to another
+        /// thread should not be an issue for most use-cases.
+        ///
+        /// However, there are cases where this [core::marker::Send] implementation might be
+        /// invalid, for example if an MMIO block was created for a core-local private address.
+        ///
+        /// In that case, it it is recommended to wrap the MMIO block in a newtype structure
+        /// and [un-implement Send](https://doc.rust-lang.org/nomicon/send-and-sync.html).
+        unsafe impl core::marker::Send for #wrapper_ident<'_> {}
```

Comment:
I wonder if the following is slightly more appropriate?
```suggestion
        unsafe impl core::marker::Send for #wrapper_ident<'_> where for<'a> &'a mut $ident: Send {}
```
That way if the wrapped struct is not `Send`, we don't accidentally enable `Send` arbitrarily for structs that aren't `Send`  themselves.

The simpler variant of 
```suggestion
        unsafe impl core::marker::Send for #wrapper_ident<'_> where $ident: Send {}
```
would also work the same, given `&mut T` is `Send` when `T: Send`, but I feel like it models it slightly more correctly given we store a mut pointer.

Reply:
I added this in addition to two tests.